central/processlisteningonport/datastore/datastore_impl_test.go

```diff
@@ -2132,3 +2142,333 @@ func (suite *PLOPDataStoreTestSuite) TestAddPodUids() {
+				newIndicatorWithDeploymentAndPod(fixtureconsts.ProcessIndicatorID3, 1*time.Hour, fixtureconsts.Deployment6, fixtureconsts.PodUID3),
+			},
+			deployments:           set.NewFrozenStringSet(fixtureconsts.Deployment6),
+			pods:                  set.NewFrozenStringSet(fixtureconsts.PodUID1, fixtureconsts.PodUID2, fixtureconsts.PodUID3),
+			expectedPlopDeletions: nil,
+		},
+		{
+			name: "some pods separate deployments - remove some indicators",
+			initialProcesses: []*storage.ProcessIndicator{
+				newIndicatorWithDeploymentAndPod(fixtureconsts.ProcessIndicatorID1, 1*time.Hour, fixtureconsts.Deployment6, fixtureconsts.PodUID1),
+				newIndicatorWithDeploymentAndPod(fixtureconsts.ProcessIndicatorID2, 20*time.Minute, fixtureconsts.Deployment5, fixtureconsts.PodUID2),
+				newIndicatorWithDeploymentAndPod(fixtureconsts.ProcessIndicatorID3, 1*time.Hour, fixtureconsts.Deployment3, fixtureconsts.PodUID3),
+			},
+			initialPlops: []*storage.ProcessListeningOnPortStorage{
+				fixtures.GetPlopStorage1(),
+				fixtures.GetPlopStorage2(),
+				fixtures.GetPlopStorage3(),
+			},
+			deployments:           set.NewFrozenStringSet(fixtureconsts.Deployment3),
+			pods:                  set.NewFrozenStringSet(fixtureconsts.PodUID3),
+			expectedPlopDeletions: []string{fixtureconsts.PlopUID1},
+		},
+		{
+			name: "some pods same deployment - remove some indicators",
+			initialProcesses: []*storage.ProcessIndicator{
+				newIndicatorWithDeploymentAndPod(fixtureconsts.ProcessIndicatorID1, 1*time.Hour, fixtureconsts.Deployment6, fixtureconsts.PodUID1),
+				newIndicatorWithDeploymentAndPod(fixtureconsts.ProcessIndicatorID2, 20*time.Minute, fixtureconsts.Deployment6, fixtureconsts.PodUID2),
+				newIndicatorWithDeploymentAndPod(fixtureconsts.ProcessIndicatorID3, 1*time.Hour, fixtureconsts.Deployment6, fixtureconsts.PodUID3),
+			},
+			initialPlops: []*storage.ProcessListeningOnPortStorage{
+				fixtures.GetPlopStorage1(),
+				fixtures.GetPlopStorage2(),
+				fixtures.GetPlopStorage3(),
+			},
+			deployments:           set.NewFrozenStringSet(fixtureconsts.Deployment6),
+			pods:                  set.NewFrozenStringSet(fixtureconsts.PodUID3),
+			expectedPlopDeletions: []string{fixtureconsts.PlopUID1},
+		},
+	}
+	for _, c := range cases {
+		suite.T().Run(c.name, func(t *testing.T) {
+			suite.TearDownTest()
+			suite.SetupTest()
+			// Add deployments if necessary
+			deploymentDS, err := deploymentStore.GetTestPostgresDataStore(suite.T(), suite.postgres.DB)
+			suite.Nil(err)
+			for _, deploymentID := range c.deployments.AsSlice() {
+				suite.NoError(deploymentDS.UpsertDeployment(suite.hasAllCtx, &storage.Deployment{Id: deploymentID, ClusterId: fixtureconsts.Cluster1}))
+			}
+
+			for _, podID := range c.pods.AsSlice() {
+				insertPod := fmt.Sprintf("INSERT INTO pods (id, clusterid) VALUES ('%s', '%s')", podID, fixtureconsts.Cluster1)
```

Comment:
It was not possible to use the pod store here as that would have created cyclical imports.